nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/chronicle/ChronicleLogManager.java

```diff
@@ -0,0 +1,206 @@
+        File path = new File(basePath.toFile(), name);
+        //noinspection ConstantConditions
+        return path.isDirectory() && path.list().length > 0;
+    }
+
+    @Override
+    public void create(String name, int size) {
+        ChronicleLogAppender cq = ChronicleLogAppender.create(new File(basePath.toFile(), name), size, retentionDuration);
+        try {
+            cq.close();
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot create and close " + name, e);
+        }
+    }
+
+    @Override
+    public boolean delete(String name) {
+        File path = new File(basePath.toFile(), name);
+        if (path.isDirectory()) {
+            deleteQueueBasePath(path);
+            return true;
+        }
+        return false;
+    }
+
+    protected LogLag getLagForPartition(String name, int partition, String group) {
+        long pos;
+        File path = new File(basePath.toFile(), name);
+        try (ChronicleLogOffsetTracker offsetTracker = new ChronicleLogOffsetTracker(path.toString(), partition, group)) {
+            pos = offsetTracker.readLastCommittedOffset();
+        }
+        ChronicleLogAppender appender = (ChronicleLogAppender) getAppender(name);
+        if (pos == 0) {
+            pos = appender.firstOffset(partition);
+        }
+        long end = appender.endOffset(partition);
+        long lag = appender.countMessages(partition, pos, end);
+        long firstOffset = appender.firstOffset(partition);
+        long endMessages = appender.countMessages(partition, firstOffset, end);
+        return new LogLag(pos, end, lag, endMessages);
+    }
+
+    @Override
+    public List<LogLag> getLagPerPartition(String name, String group) {
+        int size = getAppender(name).size();
+        List<LogLag> ret = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            ret.add(getLagForPartition(name, i, group));
+        }
+        return ret;
+    }
+
+    @Override
+    public String toString() {
+        return "ChronicleLogManager{" +
+                "basePath=" + basePath +
+                ", retentionDuration='" + retentionDuration + '\'' +
+                '}';
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Override
+    public List<String> listAll() {
+        if (!basePath.toFile().exists() || !basePath.toFile().isDirectory()) {
+            throw new IllegalArgumentException("Invalid base path: " + basePath);
+        }
+        return Arrays.asList(basePath.toFile().list((dir, name) -> new File(dir, name).isDirectory()));
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Override
+    public List<String> listConsumerGroups(String name) {
+        File mqRoot = new File(basePath.toFile(), name);
+        if (!exists(name)) {
+            throw new IllegalArgumentException("Unknown Log: " + name);
+        }
+        return Arrays.stream(mqRoot.list((dir, rep) -> new File(dir, rep).isDirectory() &&
```

Comment:
Same here with `Files#list`

Reply:
thanks I will note that I need to improve file access with Files#methods but I will not change this now, this code base has been tested for month and it sounds too risky for no gain.

Reply:
Changing to `Files` methods is not risky, it's just a cleanup to use modern APIs.